File: blind-nav-web/src/types/notification.ts
export type NotificationType = 'CRITICAL_ALERT' | 'SLA_WARNING' | 'SLA_BREACH' | 'ESCALATION' | 'GENERAL';
export type NotificationChannel = 'EMAIL' | 'SMS' | 'IN_APP' | 'PUSH';
export type NotificationStatus = 'PREPARED' | 'QUEUED' | 'SENT' | 'DELIVERED';

export interface Notification {
  id: string;
  type: NotificationType;
  channel: NotificationChannel;
  recipient: string;
  subject: string;
  body: string;
  status: NotificationStatus;
  alertId?: string;
  requestId?: string;
  createdAt: string;
  sentAt?: string;
}

export interface NotificationStats {
  total: number;
  prepared: number;
  sent: number;
  byChannel: { EMAIL: number; SMS: number; IN_APP: number; PUSH: number };
}

export interface KnowledgeArticle {
  id: string;
  title: string;
  sop_ref: string;
  content: string;
  resolution_time: string;
  categories: string[];
  relevance_score: number;
}

File: blind-nav-backend/app/schemas/notification.py
from pydantic import BaseModel
from typing import Optional


class Notification(BaseModel):
    id: str
    type: str       # CRITICAL_ALERT | SLA_WARNING | SLA_BREACH | ESCALATION | GENERAL
    channel: str    # EMAIL | SMS | IN_APP | PUSH
    recipient: str  # Role / person name
    subject: str
    body: str
    status: str     # PREPARED | QUEUED | SENT | DELIVERED
    alertId: Optional[str] = None
    requestId: Optional[str] = None
    createdAt: str
    sentAt: Optional[str] = None
